kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -637,6 +648,17 @@ private void commitFile(String encodedPartition) {
       RecordWriter writer = writers.get(encodedPartition);
       // Commits the file and closes the underlying output stream.
       writer.commit();
+
+      try {
+        String filename = getCommitFilename(encodedPartition);
+        continuumProducer.produce(
+                filename,
```

Comment:
It does seem a bit odd to have the filename in both the key and the value. But I'm not confident enough about how keys are used to say if that's a problem. My understanding is that keys are used to help with ordering and duplicate detection within a partition. So filename seems like a good key, I'm just wondering if it should also be in the value or not.

Reply:
Chris Pawling and I had a similar conversation, as he also thought it was weird. I don't think it will cause problems (I did some research into what the keys are for just now), but it is weird to duplicate the information. I kind of like having it in the value because it makes it easier to change the key in the future if we ever need to. I also like having the value of the message include everything clients need. Space is cheap and it's a bit of a wash, and I've already told Chris and Kelsey that the filename will be in the value, so how do you feel about leaving the filename in both the key and the value? 

I don't feel strongly if you'd rather we remove it from the value. I prototyped null keys just now too, but I'd be worried about a given table's messages getting out of order if we had a null key.

Reply:
Yeah, I'm fine with leaving it in both.